src/structures/Message.js

```diff
@@ -418,6 +418,22 @@ class Message extends Base {
     );
   }
 
+  /**
+   * Whether the message is crosspostable by the client user
+   * @type {boolean}
+   * @readonly
+   */
+  get crosspostable() {
+    return (
+      this.channel.type === 'news' &&
+      !this.flags.has(MessageFlags.FLAGS.CROSSPOSTED) &&
+      !this.flags.has(MessageFlags.FLAGS.IS_CROSSPOST) &&
+      this.author.id === this.client.user.id &&
```

Comment:
can't you publish other users messages? or are bots restricted to their own?

Reply:
You can indeed crosspost someone else's message if you have `MANAGE_MESSAGES`. [docs](https://discord.com/developers/docs/resources/channel#crosspost-message)

Reply:
Indeed, I'll update it to support that